### PR TITLE
Deprecate implicit default DAG schedule

### PR DESCRIPTION
**Pull request by uranusjr**
Currently, if a DAG is created without the 'schedule' argument, Airflow sets the DAG to run daily. This is almost never what people want, and every best practice tells you to always specify the argument explicitly, but we couldn't change the default due to compatibility considerations.

We can finally change this in Airflow 3.0, but first we need to tell users about it.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

**Review by jscheffl — APPROVED** [review]

File: airflow/models/dag.py [hunk 1]
```diff
@@ -678,6 +678,12 @@ def __init__(
             self.timetable = DatasetTriggeredTimetable(DatasetAll(*schedule))
             self.schedule_interval = self.timetable.summary
         elif isinstance(schedule, ArgNotSet):
+            warnings.warn(
```
Comment (jscheffl) [inline airflow/models/dag.py:683]: Just curious - maybe lagging history: Why was it never considered setting `schedule=None` per default?
If I take a look to ALL your changes in all tests I see this seems the be actually a common and reasonable default (whereas I acknowledge thatchanging the default in a breaking change might confuse and irritate a LOT of users)

Comment (jedcunningham) [reply airflow/models/dag.py:683]: We've definitely considered it and wanted to do it for a long time now. But yeah, breaking change...

**Review by jedcunningham — APPROVED** [review]

File: airflow/models/dag.py (hunk 1 above)
Comment (uranusjr) [reply airflow/models/dag.py:683]: Yeah there is already a default, and we can’t really just change that default unless we warn everyone not use that default first. I do wonder if there’s a better way.

**Review by vincbeck — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]

File: airflow/models/dag.py (hunk 1 above)
Comment (jscheffl) [reply airflow/models/dag.py:683]: Yeah... no better way... only chance would be to drop the attribute (which we did in the past once :-D) and add a new one with a new default... yeah. sad.

**Comment by kaxil** [comment]
@uranusjr @ephraimbuddy - Do we need to get this for 2.10.0rc1?

**Comment by uranusjr** [comment]
I’m going to do this another way to not break the tests for now. We still need to eventually fix all the tests when we migrate to 3.0, but it’ll buy us some time.

**Comment by uranusjr** [comment]
Alternative PR posted https://github.com/apache/airflow/pull/41397

**Comment by potiuk** [comment]
https://github.com/astronomer/airflow/pull/1512/ -should fix those warnings.

**Review by potiuk — APPROVED** [review]
LGTM but some conflicts need to be resolved.

**Comment by uranusjr** [comment]
This should be ready if CI passes.

**A backport to the 2.10 branch is needed.**

**Comment by potiuk** [comment]
> A backport to the 2.10 branch is needed.

According to the new rule it's on the committer who merges it to backport it and create a new PR if I read them right.

**Comment by eladkal** [comment]
> > A backport to the 2.10 branch is needed.
> 
> According to the new rules it's on the committer who merges it to backport it and create a new PR if I read them right.

I raised https://github.com/apache/airflow/pull/41457 to document the protocols

**Comment by uranusjr** [comment]
Finally… now merge and work on bacporting.

**Comment by malthe** [comment]
s/with/without a ‘schedule’ argument.

**Pull request merged**